src/components/resources/Sidebar/SidebarItem.tsx

```diff
@@ -1,11 +1,29 @@
 import "./Sidebar.style.scss";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { CircleIcon } from "../../core";
 import type { SidebarItemProps } from "./Sidebar.model";
 
+interface RecursiveSidebarItemProps extends SidebarItemProps {
```

Comment:
You can import the SidebarItem type on line 4 to avoid creating a new type

Reply:
resolved: [53826fd](https://github.com/CarletonComputerScienceSociety/resources/pull/6/commits/53826fd65f51b579b7001c4b65a674fec8cdb109)